ci/build_common.sh

```diff
@@ -6,108 +6,176 @@ set -eo pipefail
     exit 1
 }
 
-# Check for extra options
-# While there are more than 3 arguments, parse switches/options
-while [ "$#" -gt 3 ]
-do
-  case "${1}" in
-  -h)     usage ;;
-  -help)  usage ;;
-  --help) usage ;;
-  --verbose)           VERBOSE=1; shift ;;
-  -v)                  VERBOSE=1; shift ;;
-  -nvcc)               CUDA_COMPILER="${2}"; shift 2;;
-  -disable-benchmarks) ENABLE_CUB_BENCHMARKS="false"; shift ;;
-  *) usage ;;
-  esac
+# Parse options
+
+# Copy the args into a temporary array, since we will modify them and
+# the parent script may still need them.
+args=("$@")
+echo "Args: ${args[@]}"
+while [ "${#args[@]}" -ne 0 ]; do
+    case "${args[0]}" in
+    -v | --verbose) VERBOSE=1; args=("${args[0]:1}");;
+    -cxx)  HOST_COMPILER="${args[1]}"; args=("${args[@]:2}");;
+    -std)  CXX_STANDARD="${args[1]}";  args=("${args[@]:2}");;
+    -cuda) CUDA_COMPILER="${args[1]}"; args=("${args[@]:2}");;
+    -arch) CUDA_ARCHS="${args[1]}";    args=("${args[@]:2}");;
+    -disable-benchmarks) ENABLE_CUB_BENCHMARKS="false"; args=("${args[@]:1}");;
+    -h | -help | --help) usage ;;
+    *) echo "Unrecognized option: ${args[0]}"; usage ;;
+    esac
 done
 
-if [ $VERBOSE ]; then
-    set -x
+# Convert to full paths:
+HOST_COMPILER=$(which ${HOST_COMPILER})
+CUDA_COMPILER=$(which ${CUDA_COMPILER})
+
+GLOBAL_CMAKE_OPTIONS=""
+if [[ -n "${CUDA_ARCHS}" ]]; then
+    GLOBAL_CMAKE_OPTIONS+="-DCMAKE_CUDA_ARCHITECTURES=${CUDA_ARCHS} "
 fi
 
-if [ "$#" -ne 3 ]; then
-    echo "Invalid number of arguments"
-    usage
+if [ $VERBOSE ]; then
+    set -x
 fi
 
 # Begin processing unsets after option parsing
 set -u
 
-# Assign command line arguments to variables
-readonly HOST_COMPILER=$(which $1)
-readonly CXX_STANDARD=$2
-
-# Replace spaces, commas and semicolons with semicolons for CMake list
-readonly GPU_ARCHS=$(echo $3 | tr ' ,' ';')
 readonly PARALLEL_LEVEL=${PARALLEL_LEVEL:=$(nproc)}
 
-if [ -z ${DEVCONTAINER_NAME+x} ]; then
-    BUILD_DIR=../build/local
-else
-    BUILD_DIR=../build/${DEVCONTAINER_NAME}
+if [ -z ${CCCL_BUILD_INFIX+x} ]; then
+    CCCL_BUILD_INFIX=""
 fi
 
+# Presets will be configured in this directory:
+BUILD_DIR="../build/${CCCL_BUILD_INFIX}"
+
 # The most recent build will always be symlinked to cccl/build/latest
 mkdir -p $BUILD_DIR
 rm -f ../build/latest
 ln -sf $BUILD_DIR ../build/latest
 
-COMMON_CMAKE_OPTIONS="
-    -DCMAKE_BUILD_TYPE=Release \
-    -DCMAKE_CXX_STANDARD=${CXX_STANDARD} \
-    -DCMAKE_CUDA_STANDARD=${CXX_STANDARD} \
-    -DCMAKE_CXX_COMPILER=${HOST_COMPILER} \
-    -DCMAKE_CUDA_COMPILER=${CUDA_COMPILER} \
-    -DCMAKE_CUDA_HOST_COMPILER=${HOST_COMPILER} \
-    -DCMAKE_CUDA_ARCHITECTURES=${GPU_ARCHS} \
-    -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
-"
+# Now that BUILD_DIR exists, use readlink to canonicalize the path:
+BUILD_DIR=$(readlink -f "${BUILD_DIR}")
+
+# Prepare environment for CMake:
+export CMAKE_BUILD_PARALLEL_LEVEL="${PARALLEL_LEVEL}"
+export CTEST_PARALLEL_LEVEL="1"
+export CXX="${HOST_COMPILER}"
```

Comment:
question (for self): Does this overwrite the CXX envvar after running the script?